File: kotlin/fizzbuzz.kt

fun fizzbuzzer(num: Int, replacements: Map<Int, String>) : String =
    replacements.filter { 
        (multiple, _) -> num % multiple == 0 
    }.values.toList().joinToString("").ifEmpty{ 
        "$num" 
    }

fun fizzbuzzRange(start: Int, end: Int, replacements: Map<Int, String>) : String =
    (start..end).map { num -> 
        fizzbuzzer(num, replacements)
    }.joinToString("\n")

fun main() {
    println(fizzbuzzRange(1, 100, mapOf(
        3 to "Fizz",
        5 to "Buzz"
    )))
}

File: javascript/fizzbuzz.js
'use strict';

function fizzbuzzer(num, replacements) {
    var out = "";
    for (var multiple in replacements) {
        if (num % multiple == 0)
        out += replacements[multiple];
    }
    if (out === "") {
        return num;
    }
    return out;
}

function fizzbuzzRange(start, end, replacements) {
    var out = "";    
    for (var num = start; num < end; num++) {
        out += fizzbuzzer(num, replacements) + "\n";
    }
    return out;
}

console.log(fizzbuzzRange(1, 100, {
    3: "Fizz",
    5: "Buzz"
}));
